core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java

```diff
@@ -623,6 +623,13 @@ && exemptedFieldBySuperType(getType(variableTree), state)) {
           if (variableTree.getName().contentEquals("this")) {
             return null;
           }
+          // Ignore if parameter is part of canonical record constructor; tree does not seem
+          // to contain usage in that case, but parameter is always used implicitly
+          // For compact canonical constructor parameters don't have record flag so need to
+          // check constructor flags (`symbol.owner`) instead
+          if (hasRecordFlag(symbol) || hasRecordFlag(symbol.owner)) {
```

Comment:
Are there any cases where `hasRecordFlag(symbol)` is true but `hasRecordFlag(symbol.owner)` isn't?

Reply:
Are you asking if this could be simplified to only check `hasRecordFlag(symbol.owner)`?
It looks like the tests still pass if I locally remove the additional `hasRecordFlag(symbol)` check, should I change this then?

Reply:
> Are you asking if this could be simplified to only check `hasRecordFlag(symbol.owner)`?

Yes

> It looks like the tests still pass if I locally remove the additional `hasRecordFlag(symbol)` check, should I change this then?

I think so, unless you suspect there might be cases where checking both is necessary?